caffe2/CMakeLists.txt

```
@@ -282,6 +282,7 @@ if(NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
     add_library(tensorpipe_agent "${TORCH_SRC_DIR}/csrc/distributed/rpc/tensorpipe_agent.cpp" "${TORCH_SRC_DIR}/csrc/distributed/rpc/tensorpipe_agent.h")
     target_link_libraries(tensorpipe_agent PRIVATE torch c10d tensorpipe fmt::fmt-header-only)
     add_dependencies(tensorpipe_agent torch c10d tensorpipe)
+    target_compile_definitions(tensorpipe_agent PUBLIC USE_TENSORPIPE)
```

Shouldn't this be guarded by `USE_TENSORPIPE`?
```suggestion
    if(USE_TENSORPIPE)
      add_library(tensorpipe_agent "${TORCH_SRC_DIR}/csrc/distributed/rpc/tensorpipe_agent.cpp" "${TORCH_SRC_DIR}/csrc/distributed/rpc/tensorpipe_agent.h")
      target_link_libraries(tensorpipe_agent PRIVATE torch c10d tensorpipe fmt::fmt-header-only)
      add_dependencies(tensorpipe_agent torch c10d tensorpipe)
      target_compile_definitions(tensorpipe_agent PUBLIC USE_TENSORPIPE)
    endif()
```

---

Good catch, done! (In a slightly different way, as it's more convenient to have the `tensorpipe_agent` target define even when we don't use TensorPipe, even if it's empty).